src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -13,39 +15,55 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer;
 // but we did not want to migrate them all at once
 public class LoggerAdapter : IRazorLogger
 {
-    private readonly ILogger _logger;
+    private readonly IEnumerable<ILogger> _loggers;
     private readonly ITelemetryReporter? _telemetryReporter;
 
-    public LoggerAdapter(ILogger logger, ITelemetryReporter? telemetryReporter)
+    public LoggerAdapter(IEnumerable<ILogger> loggers, ITelemetryReporter? telemetryReporter)
     {
-        _logger = logger;
+        _loggers = loggers;
         _telemetryReporter = telemetryReporter;
     }
 
     public IDisposable BeginScope<TState>(TState state)
     {
-        return _logger.BeginScope<TState>(state);
+        var compositeDisposable = new CompositeDisposable();
+        foreach (var logger in _loggers)
+        {
+            var disposable = logger.BeginScope(state);
+            compositeDisposable.AddDisposable(disposable);
+        }
+
+        return compositeDisposable;
     }
 
     public bool IsEnabled(LogLevel logLevel)
     {
-        return _logger.IsEnabled(logLevel);
+        return _loggers.Any(l => l.IsEnabled(logLevel));
```

Comment:
This means that if any one logger says it can handle debug, then all loggers will get messages, even if they only handle warning. This then means every logger needs to check the log level each time. Seems like something people could easily forget.

Perhaps this logger should check `IsEnabled` inside each `LogXX` method?

Reply:
I thought the Loggers were supposed to be doing that within their own `LogXX` methods but it couldn't hurt to be extra defensive here, will have the LoggerAdapter check first.